src/puppetbridge.ts

```diff
@@ -671,10 +727,25 @@ export class PuppetBridge extends EventEmitter {
 		}
 	}
 
+	public trackConnectionStatus(puppetId: number, isConnected: boolean) {
+		if (Boolean(this.connectionMetricStatus[puppetId]) === isConnected) {
+			return;
+		}
+		this.connectionMetricStatus[puppetId] = isConnected;
+		if (isConnected) {
+			this.metrics.connected.inc({protocol: this.protocol.id});
+		} else {
+			this.metrics.connected.dec({protocol: this.protocol.id});
+		}
+	}
+
 	/**
 	 * Send a status message either to the status message room or to a specified room
 	 */
-	public async sendStatusMessage(puppetId: number | IRemoteRoom, msg: string) {
+	public async sendStatusMessage(puppetId: number | IRemoteRoom, msg: string, isConnected: boolean | null = null) {
+		if (isConnected !== null && typeof puppetId === "number") {
```

Comment:
```suggestion
		if (isConnected !== null) {
			this.trackConnectionStatus(typeof puppetId === "number" ? puppetId : puppetId.puppetId, isConnected);
```

Reply:
see https://github.com/Sorunome/mx-puppet-bridge/pull/72/files#diff-4cd484140d4c36875ceef0af75412976a1090961b49b5137435995661f73cb44R744 - added it to the commit